src/special_arrays/special_arrays.jl

```diff
@@ -29,6 +29,18 @@ const ImputingMatrix{T, U} = Union{PreImputingMatrix{T, U}, PostImputingMatrix{T
 const PreImputingDense = Dense{T, <: PreImputingMatrix} where T
 const PostImputingDense = Dense{T, <: PostImputingMatrix} where T
 
+# so we error on integers with missings
+Flux.onecold(y::MaybeHotArray{Maybe{T}}, labels = 1:size(y, 1)) where T<:Integer = 
+    throw(ArgumentError("MaybeHotArray{Union{T, Missing}} where T <:Integer can't produce onecold encoding, use maybecold instead."))
+# but we don't error on maybehot which has only integers
```

Comment:
Perhaps

```julia
# so we error on integers with missings  (but not on maybehot which has only integers)
 Flux.onecold(X::MaybeHotArray{<:Maybe{Integer}}, labels = 1:size(X, 1))  = 
     throw(ArgumentError("$(typeof(X)) can't produce `onecold` encoding, use `maybecold` instead."))
```

Reply:
I think this is not correct, because then this happens:
```
julia>     t2 = maybehotbatch(1:3, 1:10)
10×3 MaybeHotMatrix with eltype Bool:
 1  ⋅  ⋅
 ⋅  1  ⋅
 ⋅  ⋅  1
 ⋅  ⋅  ⋅
 ⋅  ⋅  ⋅
 ⋅  ⋅  ⋅
 ⋅  ⋅  ⋅
 ⋅  ⋅  ⋅
 ⋅  ⋅  ⋅
 ⋅  ⋅  ⋅

julia>     t2 |> typeof
MaybeHotMatrix{UInt32, Int64, Bool}

julia>            Flux.onecold(t2)
ERROR: LoadError: ArgumentError: MaybeHotMatrix{UInt32, Int64, Bool} can't produce `onecold` encoding, use `maybecold` instead.
Stacktrace:
 [1] onecold(X::MaybeHotMatrix{UInt32, Int64, Bool}, labels::UnitRange{Int64}) (repeats 2 times)
   @ Mill c:\Projects\others\Mill.jl\src\special_arrays\special_arrays.jl:33
 [2] top-level scope
   @ c:\Projects\others\Mill.jl\test\maybe_hot.jl:276
in expression starting at c:\Projects\others\Mill.jl\test\maybe_hot.jl:276
```
and also the MaybeHot with integers which does not have missing is being dispatched.